### ISSUE TITLE
Linux ARM64 support?

### ISSUE THREAD

**Issue by matrunchyk** [GoogleChromeLabs/chrome-for-testing#1]
Hi, do you plan to add linux arm64 support?

I tried https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/113.0.5672.63/linux64/chrome-linux64.zip, but it seems like it's intended for x86_64 platform:

> qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory

There is no `/lib64` directory in linux arm64, but  just `/lib`.

Thanks!

**Comment by matrunchyk** [comment]
cc @mathiasbynens

**Comment by mathiasbynens** [comment]
**Note:** This issue tracker is only for the code that lives in this repository, i.e. the dashboard + CLI scripts.

> do you plan to add linux arm64 support?

This is on our radar, and the relevant (unfortunately not public) tracking bug is <https://bugs.chromium.org/p/chromium/issues/detail?id=1440652>.

**Comment by matrunchyk** [comment]
Thank you @mathiasbynens for a fast response! Are you aware of approximate schedule if the release, by a chance?

**Comment by mathiasbynens** [comment]
> By any chance, do you have an approximate schedule for the release?

No ETA. We’ll update this repository when the new binaries become available.

**Comment by thiagowfx** [comment]
Reopening to keep track of the FR but, again, the upstream is https://bugs.chromium.org/p/chromium/issues/detail?id=1440652

**Comment by diemol** [comment]
It'd be nice to have this so we can provide Docker images for ARM with Chrome instead of Chromium.

https://github.com/seleniumhq-community/docker-seleniarm

**Comment by fritzo** [comment]
@thiagowfx thanks for keeping this public issue open (the upstream issue gives me Permission Denied)

**Comment by seanpoulter** [comment]
Please let us know how we can help support this.

**Comment by devsheva** [comment]
Still no news ?

**Comment by sandstrom** [comment]
ARM64 images for Linux would be very useful!

We currently need to apply several workarounds to make it work. 

An option would be to switch to another headless browser, but if possible we'd like to stick with Chrome for testing, because overall it's really good!

**Comment by montehurd** [comment]
Would very much like this too.

**Comment by sandstrom** [comment]
Github has just released ARM runners for Github Actions:

https://github.blog/2024-06-03-arm64-on-github-actions-powering-faster-more-efficient-build-systems/

Would be neat if Chrome for testing would provide a Linux ARM64 build for use with such runners.

**Comment by solving321** [comment]
Can anybody please answer this,

I am using debian ARM64 and below code.

```
const { launch } = await import('chrome-launcher');

  const launchOptions = {
    startingUrl: website_url,
    chromeFlags: [
      // '--headless',
      '--start-maximized',
      '--no-sandbox',
      '--disable-gpu',
      '--hide-scrollbars',
      '--disable-web-security',
      '--ignore-certificate-errors',
      '--no-first-run',
      '--no-default-browser-check',
      '--disable-infobars',
      '--disable-background-timer-throttling',
      '--disable-features=TranslateUI',
      '--disable-background-networking',
      '--disable-dev-shm-usage',
      '--safebrowsing-disable-auto-update',
      '--disable-client-side-phishing-detection',
      '--disable-sync'
    ]
  };
  
  const chrome = await launch(launchOptions);

  console.log(`Chrome debugging port running on ${chrome.port}`);
  const client = await CDP({ port: chrome.port });                      // this line is throwing error;
  console.log('CDP connected client: ', client);
  await client.Page.enable();
  ```
  
  
  
  Now, 
  
` const client = await CDP({ port: chrome.port });`
  
  is not working and throwing error:
  
 ```
Error: connect ECONNREFUSED 127.0.0.1:59742
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 59742
}
 ```
 
 Please help.

**Comment by mathiasbynens** [comment]
Unfortunately Chrome does not officially support `linux-arm64` at this time. We can only make Chrome for Testing available on platforms that are officially supported by the Chrome product.

**Comment by sandstrom** [comment]
@mathiasbynens Makes sense, but would you be willing to keep it open?

It's the by far most upvoted issue in this tracker:

https://github.com/GoogleChromeLabs/chrome-for-testing/issues?q=is%3Aissue+sort%3Areactions-%2B1-desc+

And it will very likely be supported eventually anyway (it's not like ARM is going away anytime soon).

**Comment by solving321** [comment]
@benschwarz 

Not only that, I have installed UTM on mac to test something on Debian, ARM64 version is only compatible with mac's recent version. Unfortunately it don't have chrome support so I have to install chromium to test.

**Comment by mathiasbynens** [comment]
> In terms of browser automation, Chrome is heavily used for by a large community of people. By using ARM64, those automation workloads could be run on lower wattage hardware, with less heat and emissions output, at a lower cost.

FWIW, I fully agree and am sympathetic to this cause.

> Do you think it's worth opening a CR bug to discuss the addition of linux-arm64 releases for Chrome?

For CfT specifically, there was an issue at https://crbug.com/40909440 which unfortunately resulted in [the outcome described in my earlier comment](https://github.com/GoogleChromeLabs/chrome-for-testing/issues/1#issuecomment-2419285549).

I encourage you to [file a separate Chromium issue](https://crbug.com/new) **requesting `linux-arm64` support for all of Chrome** so people can star it / subscribe to it for updates. If/when the product decision is made to support `linux-arm64` we’d happily provide `linux-arm64` CfT builds. Until that decision is made it is however not feasible for us to do anything on the CfT side.

**Comment by Niek** [comment]
FWIW Microsoft releases `playwright` docker images for arm64, which contain native arm64 Chromium/Firefox/WebKit. See e.g. `mcr.microsoft.com/playwright:v1.48.1-arm64`.

**Comment by benschwarz** [comment]
@mathiasbynens, thanks to your recommendation, I have logged an issue on Chromium.

Everyone who wants to support Chrome arm64:  [Go star this issue](https://issues.chromium.org/issues/374811603)

**Comment by stephankaag** [comment]
Should not be too big of an effort. 

I’m already compiling arm64 binaries myself for usage on AWS Lambda. 

https://github.com/chromium-for-lambda/chromium-binaries

**Comment by mathiasbynens** [comment]
> @mathiasbynens, thanks to your recommendation, I have logged an issue on Chromium.
> 
> Everyone who wants to support Chrome arm64: [Go star this issue](https://issues.chromium.org/issues/374811603)

Thank you! I’ve added a link to the issue to the top post here as well, so hopefully more people can see and star it.

**Comment by slhck** [comment]
FYI the linked issue has been marked as "Won't fix" in November: https://issues.chromium.org/issues/374811603#comment10

**Comment by dennisoelkers** [comment]
Can anyone from Google help us understand why this is such a hard thing to do, when everyone is already building their own arm64 binaries? Is it really just because of _the graphical end-user browser_ not supported _officially_ on linux-arm64?

**Comment by OrKoN** [comment]
> Can anyone from Google help us understand why this is such a hard thing to do, when everyone is already building their own arm64 binaries? Is it really just because of _the graphical end-user browser_ not supported _officially_ on linux-arm64?

See https://g-issues.chromium.org/issues/374811603#comment10

**Comment by sandstrom** [comment]
Myself and probably quite a few others asking for this use it for automation, via e.g. Playwright (https://playwright.dev/).

Perhaps it would make more sense to see if we can get one of the other major engines (Firefox or Safari) to provide linux-arm64 builds suitable for testing? (playwright and most other automation software work with all major engines)

I don't know how far away any of them are, but maybe the architecture for one of those two projects make it such that it's less work.

cc @Skn0tt @mxschmitt

**Comment by Niek** [comment]
> Myself and probably quite a few others asking for this use it for automation, via e.g. Playwright (https://playwright.dev/).
> 
> Perhaps it would make more sense to see if we can get one of the other major engines (Firefox or Safari) to provide linux-arm64 builds suitable for testing? (playwright and most other automation software work with all major engines)
> 
> I don't know how far away any of them are, but maybe the architecture for one of those two projects make it such that it's less work.
> 
> cc @Skn0tt @mxschmitt

See my previous comment here: https://github.com/GoogleChromeLabs/chrome-for-testing/issues/1#issuecomment-2419331864 - MS already publishes an arm64 Chromium build. For example:
```bash
docker run --rm -it mcr.microsoft.com/playwright:v1.49.1-arm64 /ms-playwright/chromium-1148/chrome-linux/chrome --no-sandbox --headless=new```

Or use it in your `Dockerfile`:
```dockerfile
COPY --from=mcr.microsoft.com/playwright:v1.49.1-arm64 /ms-playwright/chromium-1148/chrome-linux/ /opt/chrome
```

**Comment by LorDisturbia** [comment]
The problem with Playwright's Chromium builds is that, AFAIU, they can't contain proprietary stuff for licensing reasons. For example, they can't contain proprietary audio and video codecs.

They're still fine for most automations, but if you're testing something that has to do with audio and video, and you want to do it on ARM, you might be out of luck.

**Comment by lino-levan** [comment]
I'm not seeing the arm linux builds on https://googlechromelabs.github.io/chrome-for-testing/known-good-versions-with-downloads.json. Was this intended to be closed?

**Comment by mathiasbynens** [comment]
> Was this intended to be closed?

Yes. See the above discussion for details.

**Comment by cAstraea** [comment]
Can you support this ? I will pay 500,000$ for open source support for this

**Comment by MikeMcC399** [comment]
Still waiting, despite reading in https://blog.chromium.org/2026/03/bringing-chrome-to-arm64-linux-devices.html

> We’re excited to announce that Google will launch Chrome for ARM64 Linux devices in Q2 2026, following the successful expansion of Chrome to Arm-powered macOS devices in 2020 and Arm-powered Windows devices in 2024.

and now we're in Q3 2026 already with no new launch date.

**Comment by mathiasbynens** [comment]
Hi folks, the public tracking issue for the linux-arm64 release is here: https://crbug.com/374811603 On that ticket, I see that the target date moved from “end of Q2” to “July”.

**Comment by MikeMcC399** [comment]
I'm not sure how reliable the "Targeted To 2026-07" in https://crbug.com/374811603 is. The comments in the tracker are more vague:

April 2026
> This is still in progress. We’ve done a bit of work internally. 

June 2026
> There’s a bit more infrastructure work. This will be out soon.

~~I'd like to believe that there will be a release this week though!~~ Edit: DIdn't happen

**Comment by amigrave** [comment]
Once released, will the `linux64` platform be split into `linux-amd64` and `linux-arm64` in the API results ?

cf, https://googlechromelabs.github.io/chrome-for-testing/141.0.7390.54.json

**Comment by MikeMcC399** [comment]
@mathiasbynens 

> Hi folks, the public tracking issue for the linux-arm64 release is here: https://crbug.com/374811603 On that ticket, I see that the target date moved from “end of Q2” to “July”.

It seems that the release is delayed and I don't see any new activity or new target date. Is this still planned?

**Comment by mathiasbynens** [comment]
See https://crbug.com/374811603#comment33:

> We released this publicly on 2026-07-30. You can download Chrome for `linux-arm64` from the Chrome download page at https://google.com/chrome/ (alias https://chrome.com/) and choosing the arm64 `.deb` or `.rpm` package.
> 
> Downloads are available for Google Chrome across all of our release channels. **We’re still working on publishing the Chrome for Testing variant.**

**Comment by MikeMcC399** [comment]
@mathiasbynens 

That's great news!

Could you re-open the issue, as it has been closed with "Not planned", then later close it when Chrome for Testing is available.

**Comment by mathiasbynens** [comment]
@MikeMcC399 Sure thing.

For future reference, this issue tracker is meant to cover the code in this repository (i.e. the CfT availability dashboard and the JSON API). Generally feature requests like “add support for platform XYZ” should not be filed here but rather upstream in the Chromium tracker.

Given the interest in linux-arm64 for CfT I’ll make a one-time exception to reopen this though. Once the remaining CfT builders are set up (tracked here: https://crbug.com/374811603) I’ll use this GitHub issue to post updates on updating the dashboard + JSON API, finalizing the public release. Stay tuned!

**Comment by MikeMcC399** [comment]
@mathiasbynens 

Your support is much appreciated! 👍🏻 

Do you have any planning date for Cft on Linux/arm64 that you can share? I don't see anything specific in https://crbug.com/374811603.

**Comment by mathiasbynens** [comment]
**Update:** I have prepared the necessary changes to this repository. https://github.com/GoogleChromeLabs/chrome-for-testing/pull/213 Once all builders complete and we get a complete set of CfT binaries across all platforms for a single Chrome version, I’ll land this.

**Comment by mathiasbynens** [comment]
I’ve just merged #213 and triggered a data update in 1dffbc44c4e5bbbb6ed501a4cc43f4a3936bf96e, which successfully picked up the new `linux-arm64` archives. I’ve verified that the `linux-arm64` entries [appear on the dashboard](https://googlechromelabs.github.io/chrome-for-testing/#canary) and in [the CfT API endpoints](https://github.com/GoogleChromeLabs/chrome-for-testing#json-api-endpoints).

As documented in the README, `linux-arm64` CfT builds are available starting with version **153.0.8001.0**.

Thanks everyone for your support and encouragement throughout the years! We’re super excited to get this out the door!
